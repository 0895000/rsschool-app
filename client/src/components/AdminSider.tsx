import {
  BellOutlined,
  GlobalOutlined,
  IdcardOutlined,
  MenuFoldOutlined,
  MenuUnfoldOutlined,
  OrderedListOutlined,
  QuestionOutlined,
  TeamOutlined,
  IdcardFilled,
} from '@ant-design/icons';
import { Layout, Menu } from 'antd';
import * as React from 'react';

const { Sider } = Layout;

type State = { collapsed: boolean };

class AdminSider extends React.Component<any, any> {
  state: State = {
    collapsed: true,
  };

  toggle = () => {
    this.setState({
      collapsed: !this.state.collapsed,
    });
  };

  render() {
    const menuIconProps = {
      onClick: this.toggle,
      style: { fontSize: '20px', display: 'block', lineHeight: '30px', padding: '20px 32px' },
    };
    return (
      <Sider trigger={null} collapsible collapsed={this.state.collapsed} theme="dark">
        <h4>
          <span>Admin Area</span>
          {this.state.collapsed ? <MenuUnfoldOutlined {...menuIconProps} /> : <MenuFoldOutlined {...menuIconProps} />}
        </h4>

        <Menu theme="dark" mode="inline">
          <Menu.Item key="1">
            <a href="/admin/courses">
              <GlobalOutlined />
              <span>Courses</span>
            </a>
          </Menu.Item>

          <Menu.Item key="2">
            <a href="/admin/stages">
              <OrderedListOutlined />
              <span>Stages</span>
            </a>
          </Menu.Item>

          <Menu.Item key="3">
            <a href="/admin/tasks">
              <QuestionOutlined />
              <span>Tasks</span>
            </a>
          </Menu.Item>

          <Menu.Item key="4">
            <a href="/admin/events">
              <BellOutlined />
              <span>Events</span>
            </a>
          </Menu.Item>

          <Menu.Item key="5">
            <a href="/admin/users">
              <TeamOutlined />
              <span>Users</span>
            </a>
          </Menu.Item>

          <Menu.Item key="6">
            <a href="/admin/registrations">
              <IdcardOutlined />
              <span>Registrations</span>
            </a>
          </Menu.Item>

          <Menu.Item key="7">
            <a href="/admin/mentor-registry">
              <IdcardFilled />
              <span>Mentor Registry</span>
            </a>
          </Menu.Item>
        </Menu>
      </Sider>
    );
  }
}

export { AdminSider };
