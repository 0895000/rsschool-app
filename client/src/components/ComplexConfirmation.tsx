import { Modal, Input, Typography, Divider } from 'antd';
import { ExclamationCircleOutlined } from '@ant-design/icons';
import { useState } from 'react';
import random from 'lodash/random';

type AnyFunc = (...params: any) => any;

const { Text } = Typography;

type Props = {
  onOk: AnyFunc;
  keyLength: number;
  message: string;
  isConfirmationVisible: boolean;
  hideConfirmation: AnyFunc;
};

export function ComplexConfirmation(props: Props) {
  const { onOk, keyLength, message, hideConfirmation, isConfirmationVisible } = props;

  const [isKeyMatch, setKeyMatch] = useState(false);
  let key = '';

  for (let i = 0; i < keyLength; i++) key += random(0, 9);

  const checkKeyMatch = (e: any) => {
    if (e.target.value === key) {
      setKeyMatch(true);
    } else {
      setKeyMatch(false);
    }
  };

  const title = (
    <>
      <ExclamationCircleOutlined /> Are you sure?
    </>
  );

  return (
    <Modal
      title={title}
      centered={true}
      onOk={onOk}
      visible={isConfirmationVisible}
      onCancel={() => {
        hideConfirmation();
      }}
      closable={false}
      okButtonProps={{ disabled: !isKeyMatch }}
    >
      <Text underline strong>
        {message}
      </Text>
      <Divider plain>
        Enter following number to confirm action: <Text strong>{key}</Text>
      </Divider>
      <Input placeholder="Enter the number" type="text" onChange={checkKeyMatch} />
    </Modal>
  );
}
