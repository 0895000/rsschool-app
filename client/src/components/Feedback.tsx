import * as React from 'react';
import {List} from 'antd';

const publicRoutes = [
  {
    name: `👏 Say Thank you (Discord >> #gratitude)`,
    link: `/gratitude`,
    newTab: false,
  },
  {
    name: `💌 Feedback on Student/Mentor`,
    link: `/private-feedback`,
    newTab: false,
  },
  {
    name: `📝 Feedback on RS School`,
    link: `https://docs.google.com/forms/d/1F4NeS0oBq-CY805aqiPVp6CIrl4_nIYJ7Z_vUcMOFrQ/viewform`,
    newTab: true,
  },
];

type LinkInfo = { name: string; link: string; newTab: boolean };

class Feedback extends React.Component<any, any> {
  render() {
    return (
      <div className="mt-5">
        <h4>Feedback</h4>
      <List
        bordered
        size="small"
        dataSource={publicRoutes}
        renderItem={(linkInfo: LinkInfo) => (
          <List.Item key={linkInfo.link}>
            <a target={linkInfo.newTab ? '_blank' : '_self'} href={linkInfo.link}>
              {linkInfo.name}
            </a>
          </List.Item>
        )}
      />
      </div>
    );
  }
}

export { Feedback };
