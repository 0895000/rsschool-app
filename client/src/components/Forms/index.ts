export * from './GdprCheckbox';
export * from './ScoreInput';
export * from './CommentInput';
