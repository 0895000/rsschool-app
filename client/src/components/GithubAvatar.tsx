import * as React from 'react';
import { Avatar } from 'antd';

type Props = {
  githubId: string;
  size: 24 | 48 | 96;
  style?: React.CSSProperties;
};

export function GithubAvatar({ githubId, size, style }: Props) {
  return <Avatar src={`https://github.com/${githubId}.png?size=${size * 2}`} size={size} style={style} />;
}
