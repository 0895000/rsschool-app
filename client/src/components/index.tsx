export * from './AdminSider';
export * from './Feedback';
export * from './FooterLayout';
export * from './GithubAvatar';
export * from './GithubUserLink';
export * from './Header';
export * from './Help';
export * from './LoadingScreen';
export * from './LocationSelect';
export { MentorSearch } from './MentorSearch';
export * from './PageLayout';
export * from './Rating';
export * from './RegistryBanner';
export * from './SocialNetworks';
export * from './CommentModal';
export * from './StudentMentorModal';
export { StudentSearch } from './StudentSearch';
export { UserSearch } from './UserSearch';
export * from './withSession';
