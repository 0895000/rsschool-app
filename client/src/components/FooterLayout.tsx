import * as React from 'react';
import { Col, Layout, Row, Divider, Button } from 'antd';
import { Feedback } from './Feedback';
import { Help } from './Help';
import { SocialNetworks } from './SocialNetworks';

const { Footer } = Layout;

class FooterLayout extends React.Component<any, any> {
  render() {
    return (
      <div>
        <Footer className="footer-dark">
          <Row gutter={40}>
            <Col xs={24} sm={8} md={8} lg={8}>
              <Feedback />
            </Col>
            <Col xs={24} sm={8} md={8} lg={8}>
              <Help />
            </Col>
            <Col xs={24} sm={8} md={8} lg={8}>
              <SocialNetworks />
            </Col>
          </Row>
          <Divider />
          <h5> Thank you for your support! üéâ</h5>
          <p>
            <object
              type="image/svg+xml"
              data="https://opencollective.com/rsschool/backers.svg?avatarHeight=36&button=false&width=300"
            />
          </p>
          <p>
            <Button size="small" href="https://opencollective.com/rsschool#section-contribute" target="_blank" ghost>
              ‚ù§Ô∏è Make a donation
            </Button>
          </p>
          <p className="text-center">
            <small>&copy; The Rolling Scopes 2020</small>
          </p>
        </Footer>
      </div>
    );
  }
}

export { FooterLayout };
