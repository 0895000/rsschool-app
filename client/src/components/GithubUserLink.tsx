import { GithubAvatar } from './GithubAvatar';
import css from 'styled-jsx/css';

export function GithubUserLink(props: { value: string }) {
  return (
    <>
      <a className="link-user-github" href={`/profile?githubId=${props.value}`}>
        <GithubAvatar size={24} githubId={props.value} /> {props.value}
      </a>
      <style jsx>{styles}</style>
    </>
  );
}

const styles = css`
  .link-user-github {
    white-space: nowrap;
  }
`;
