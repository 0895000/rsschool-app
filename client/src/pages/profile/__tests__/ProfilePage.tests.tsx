import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { NextRouter } from 'next/router';
import { Session } from 'components/withSession';
import { ProfilePage } from '../index';
// import { GeneralInfo } from '../../../../../common/models/profile';

jest.mock('next/config', () => () => ({}));
jest.mock('services/user', () => ({
    UserService: class UserService {
      getProfileInfo() {
        return jest.fn();
      }
    },
  }),
);

describe('ProfilePage', () => {
  const profile = {
    generalInfo: {
      name: 'Dzmitry Petrov',
      githubId: 'petrov',
      aboutMyself: 'Test',
      locationName: 'Minsk',
      locationId: '1',
      educationHistory: null,
      englishLevel: 'a2+',
    },
    permissionsSettings: {
      isProfileVisible: { all: true },
      isAboutVisible: { mentor: true, student: false, all: false },
      isEducationVisible: { mentor: true, student: false, all: false },
      isEnglishVisible: { student: false, all: false },
      isEmailVisible: { student: false, all: false },
      isTelegramVisible: { student: false, all: false },
      isSkypeVisible: { student: false, all: false },
      isPhoneVisible: { student: false, all: false },
      isContactsNotesVisible: { student: true, all: false },
      isLinkedInVisible: { mentor: true, student: false, all: false },
      isPublicFeedbackVisible: { mentor: true, student: true, all: false },
      isMentorStatsVisible: { mentor: true, student: true, all: false },
      isStudentStatsVisible: { student: false, all: false },
    },
    contacts: {
      phone: '+375292123456',
      email: 'petro@gmail.com',
      skype: 'petro:live',
      telegram: 'petro',
      notes: 'discord: @petro, instagram: @petro12',
    },
    isPermissionsSettingsChanged: true,
    isProfileSettingsChanged: true,
  };
  const session = {
    id: 2020,
    githubId: 'mikhama',
    isAdmin: true,
    isHirer: false,
    isActivist: false,
    roles: {
      1: 'mentor',
      2: 'student',
      11: 'mentor',
    },
    coursesRoles: {
      13: [
        'manager',
      ],
    },
  } as Session;
  const router = {
    query: {
      githubId: 'petrov',
    },
    asPath: '/#edit/',
  } as unknown as NextRouter;

  describe('Should render correctly', () => {
    it('if full info about profile is in the state', () => {
      const wrapper = shallow(
        <ProfilePage
          session={session}
          router={router}
        />,
      );
      wrapper.setState({ profile });
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
