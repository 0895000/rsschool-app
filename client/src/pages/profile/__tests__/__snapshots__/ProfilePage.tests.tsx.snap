// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage Should render correctly if  1`] = `
<Fragment>
  <LoadingScreen
    show={true}
  >
    <Header
      isProfileEditingModeEnabled={false}
      isProfilePage={false}
      isSaveButtonVisible={false}
      onChangeProfilePageMode={[Function]}
      onSaveClick={[Function]}
      username="mikhama"
    />
    <Spin
      delay={200}
      size="default"
      spinning={false}
      wrapperClassName=""
    >
      <div
        style={
          Object {
            "padding": 10,
          }
        }
      >
        <Masonry
          breakpointCols={
            Object {
              "1100": 3,
              "500": 1,
              "700": 2,
              "default": 4,
            }
          }
          className="jsx-3803498300"
          columnClassName="jsx-110195169"
        >
          <div
            key="card-0"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <MainCard
              data={
                Object {
                  "aboutMyself": "Test",
                  "educationHistory": null,
                  "englishLevel": "a2+",
                  "githubId": "petrov",
                  "locationId": "1",
                  "locationName": "Minsk",
                  "name": "Dzmitry Petrov",
                }
              }
              isEditingModeEnabled={false}
              onPermissionsSettingsChange={[Function]}
              onProfileSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
            />
          </div>
          <div
            key="card-1"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <AboutCard
              data={
                Object {
                  "aboutMyself": "Test",
                  "educationHistory": null,
                  "englishLevel": "a2+",
                  "githubId": "petrov",
                  "locationId": "1",
                  "locationName": "Minsk",
                  "name": "Dzmitry Petrov",
                }
              }
              isEditingModeEnabled={false}
              onPermissionsSettingsChange={[Function]}
              onProfileSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
            />
          </div>
          <div
            key="card-2"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <EnglishCard
              data={
                Object {
                  "aboutMyself": "Test",
                  "educationHistory": null,
                  "englishLevel": "a2+",
                  "githubId": "petrov",
                  "locationId": "1",
                  "locationName": "Minsk",
                  "name": "Dzmitry Petrov",
                }
              }
              isEditingModeEnabled={false}
              onPermissionsSettingsChange={[Function]}
              onProfileSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
            />
          </div>
          <div
            key="card-3"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <EducationCard
              data={
                Object {
                  "aboutMyself": "Test",
                  "educationHistory": null,
                  "englishLevel": "a2+",
                  "githubId": "petrov",
                  "locationId": "1",
                  "locationName": "Minsk",
                  "name": "Dzmitry Petrov",
                }
              }
              isEditingModeEnabled={false}
              onPermissionsSettingsChange={[Function]}
              onProfileSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
            />
          </div>
          <div
            key="card-4"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <ContactsCard
              data={
                Object {
                  "email": "petro@gmail.com",
                  "notes": "discord: @petro, instagram: @petro12",
                  "phone": "+375292123456",
                  "skype": "petro:live",
                  "telegram": "petro",
                }
              }
              isEditingModeEnabled={false}
              onPermissionsSettingsChange={[Function]}
              onProfileSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
            />
          </div>
        </Masonry>
        <JSXStyle
          id="3803498300"
        >
          div.jsx-3803498300{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;margin-left:-16px;width:auto;}
        </JSXStyle>
        <JSXStyle
          id="110195169"
        >
          div.jsx-110195169{padding-left:16px;background-clip:padding-box;}
        </JSXStyle>
      </div>
    </Spin>
  </LoadingScreen>
</Fragment>
`;

exports[`ProfilePage Should render correctly if full info about profile is in the state 1`] = `
<Fragment>
  <LoadingScreen
    show={true}
  >
    <Header
      isProfileEditingModeEnabled={false}
      isProfilePage={false}
      isSaveButtonVisible={false}
      onChangeProfilePageMode={[Function]}
      onSaveClick={[Function]}
      username="mikhama"
    />
    <Spin
      delay={200}
      size="default"
      spinning={false}
      wrapperClassName=""
    >
      <div
        style={
          Object {
            "padding": 10,
          }
        }
      >
        <Masonry
          breakpointCols={
            Object {
              "1100": 3,
              "500": 1,
              "700": 2,
              "default": 4,
            }
          }
          className="jsx-3803498300"
          columnClassName="jsx-110195169"
        >
          <div
            key="card-0"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <MainCard
              data={
                Object {
                  "aboutMyself": "Test",
                  "educationHistory": null,
                  "englishLevel": "a2+",
                  "githubId": "petrov",
                  "locationId": "1",
                  "locationName": "Minsk",
                  "name": "Dzmitry Petrov",
                }
              }
              isEditingModeEnabled={false}
              onPermissionsSettingsChange={[Function]}
              onProfileSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
            />
          </div>
          <div
            key="card-1"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <AboutCard
              data={
                Object {
                  "aboutMyself": "Test",
                  "educationHistory": null,
                  "englishLevel": "a2+",
                  "githubId": "petrov",
                  "locationId": "1",
                  "locationName": "Minsk",
                  "name": "Dzmitry Petrov",
                }
              }
              isEditingModeEnabled={false}
              onPermissionsSettingsChange={[Function]}
              onProfileSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
            />
          </div>
          <div
            key="card-2"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <EnglishCard
              data={
                Object {
                  "aboutMyself": "Test",
                  "educationHistory": null,
                  "englishLevel": "a2+",
                  "githubId": "petrov",
                  "locationId": "1",
                  "locationName": "Minsk",
                  "name": "Dzmitry Petrov",
                }
              }
              isEditingModeEnabled={false}
              onPermissionsSettingsChange={[Function]}
              onProfileSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
            />
          </div>
          <div
            key="card-3"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <EducationCard
              data={
                Object {
                  "aboutMyself": "Test",
                  "educationHistory": null,
                  "englishLevel": "a2+",
                  "githubId": "petrov",
                  "locationId": "1",
                  "locationName": "Minsk",
                  "name": "Dzmitry Petrov",
                }
              }
              isEditingModeEnabled={false}
              onPermissionsSettingsChange={[Function]}
              onProfileSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
            />
          </div>
          <div
            key="card-4"
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <ContactsCard
              data={
                Object {
                  "email": "petro@gmail.com",
                  "notes": "discord: @petro, instagram: @petro12",
                  "phone": "+375292123456",
                  "skype": "petro:live",
                  "telegram": "petro",
                }
              }
              isEditingModeEnabled={false}
              onPermissionsSettingsChange={[Function]}
              onProfileSettingsChange={[Function]}
              permissionsSettings={
                Object {
                  "isAboutVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isContactsNotesVisible": Object {
                    "all": false,
                    "student": true,
                  },
                  "isEducationVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isEmailVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isEnglishVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isLinkedInVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": false,
                  },
                  "isMentorStatsVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isPhoneVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isProfileVisible": Object {
                    "all": true,
                  },
                  "isPublicFeedbackVisible": Object {
                    "all": false,
                    "mentor": true,
                    "student": true,
                  },
                  "isSkypeVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isStudentStatsVisible": Object {
                    "all": false,
                    "student": false,
                  },
                  "isTelegramVisible": Object {
                    "all": false,
                    "student": false,
                  },
                }
              }
            />
          </div>
        </Masonry>
        <JSXStyle
          id="3803498300"
        >
          div.jsx-3803498300{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;margin-left:-16px;width:auto;}
        </JSXStyle>
        <JSXStyle
          id="110195169"
        >
          div.jsx-110195169{padding-left:16px;background-clip:padding-box;}
        </JSXStyle>
      </div>
    </Spin>
  </LoadingScreen>
</Fragment>
`;
